### DISCUSSION TITLE
Interactive molecular drawings?

### DISCUSSION THREAD

**Discussion by bertiewooster** [rdkit/rdkit#5571] [Q&A] [answered]
Is there a way to draw a molecule to the screen, let a user click on some part of it (atom or bond), and then programmatically detect what part of it they clicked on?

**Comment by ptosco** [comment, marked as answer]
This is a link to a blog post that Greg published some time ago dealing with this topic:
http://rdkit.blogspot.com/2019/08/an-interactive-rdkit-widget-for-jupyter.html
I am going to add support for this in the JavaScript layer as soon as I find some time to work on it, such that it can be used in web applications.

**Reply by bertiewooster** [reply]
Great, thanks! I downloaded the Jupyter notebook in that blog post and got it to run after replacing a `d.` on a line by itself with `d.TagAtoms(dm)`. (I also submitted a [pull request](https://github.com/greglandrum/rdkit_blog/pull/10) so the file can be updated in the rdkit_blog repo). The notebook displays a molecule, I click on an atom, and I can get the selected atom by updating (re-running the cell) `w.selected_atoms`.

> I am going to add support for this in the JavaScript layer as soon as I find some time to work on it, such that it can be used in web applications.

That would be great. Until then, I'm thinking it should be possible to make this work on a web page using a Python backend such as Flask?

As a future feature idea, it might be nice to support clicking on bonds as well as atoms.
